File: src/devng/projecteuler/problem019/problem019.py
#!/usr/bin/env python
# coding: utf-8

import datetime

def count_sundays(from_year, to_year):
    """Counts how many Sundays fell on the first of the month during a given period."""
    sundays = 0
    for year in range(from_year, to_year):
        for month in range(1, 13):
            d = datetime.date(year, month, 1)
            if d.weekday() == 6:
                sundays += 1
    return sundays

def main():
    sundays = count_sundays(1901, 2001)
    print(sundays)

if __name__ == '__main__':
    main()

File: src/devng/projecteuler/problem019/problem019.go
package main

import "time"
import "fmt"

// CountSundays counts how many Sundays fell on the first of the month during a given period.
func CountSundays(fromYear int, toYear int) int {
  sundays := 0

  for year := fromYear; year < toYear; year++ {
    for month := 1; month < 13; month++ {
      t := time.Date(year, time.Month(month), 1, 0, 0, 0, 0, time.UTC)
      if t.Weekday() == time.Sunday {
        sundays++
      }
    }
  }
  return sundays
}

func main() {
  fmt.Println(CountSundays(1901, 2001))
}
